source/blog/2021-11-26-dbal-3.2.0.md

```diff
@@ -0,0 +1,140 @@
+
+### The new approach
+
+Instead of comparing abstract definitions on a per-property basis, the new implementation compares the DDL that is generated from both definitions for the target database platform. If the definitions produce the same DDL, they are considered equal. According to the tests and the number of resolved issues, this approach should be more accurate and less error-prone.
+
+Implementing this approach was impossible without introducing a new API which rendered the existing API obsolete.
+
+Prior to DBAL `3.2.0`, the schema comparator could be only instantiated directly via the `new` keyword:
+
+```php
+$comparator = new Comparator();
+```
+
+Instantiated like this, the comparator doesn't have a notion of the target database platform and cannot perform the comparison properly. That is why, this way of instantiation is deprecated in favor of instantiating the comparator by the schema manager:
+
+```php
+$schemaManager = $connection->createSchemaManager();
+$comparator = $schemaManager->createComparator();
+```
+
+This way, the schema manager can instantiate a platform-specific comparator and provide it with the necessary context (e.g. the default collation used by the database).
+
+While the old API is still available, it is recommended to use the new API for more accurate comparison.
+
+## Support for `psr/cache` ([#4620](https://github.com/doctrine/dbal/pull/4620))
+
+Since the Doctrine Cache library is being sunset, the new DBAL relase introduced the ability to use a PSR-6 compatible implementation for result caching.
+
+While both the `doctrine/cache` and `psr/cache` APIs will be supported until the next major DBAL release, we recommend users to switch to a PSR-6 compatible implementation in their projects.
+
+## Support for `psr/log` ([#4967](https://github.com/doctrine/dbal/pull/4967))
+
+The `SQLLogger` interface was designed long ago and has certain limitations: there is no way to log messages at different logger levels and it is really challenging to extend the logger functionality without introducing breaking API changes.
+
+The new DBAL release introduces a new middleware that can delegate logging to a PSR-3 compatible implementation.
+
+Note, that the new logger won't produce the messages identical to the ones produced by the old one. If you have any processes built around analysing log messages, you may need to make some changes before adopting the new API.
+
+## Always cache the full result ([#5003](https://github.com/doctrine/dbal/pull/5003))
+
+The implementation of the result cache prior to DBAL 3.2.0 would store the result set in the cache only once it was fetched completely. It led to the following issues:
+
+1. If the result isn't yet cached and its consumer didn't fetch it completely, the query would be executed again.
+2. In case of a cache miss, the DBAL would call `get()` the cache twice: once to fetch the data and once to merge the just fetched result with other results that may be stored in the cache.
```

Comment:
Note sure about that one:
```suggestion
2. In case of a cache miss, the DBAL would `get()` the cache entry twice: once to fetch the data and once to merge the just fetched result with other results that may be stored in the cache.
```